x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -199,4 +225,45 @@ export class Embeddable extends AbstractEmbeddable<LensEmbeddableInput, LensEmbe
       }
     }
   }
+
+  async initializeOutput() {
+    if (!this.savedVis) {
+      return;
+    }
+    const promises = this.savedVis.state.datasourceMetaData.filterableIndexPatterns.map(
+      async ({ id }) => {
+        try {
+          return await this.deps.indexPatternService.get(id);
+        } catch (error) {
+          // Unable to load index pattern, ignore error as the index patterns are only used to
+          // configure the filter and query bar - there is still a good chance to get the visualization
+          // to show.
+          return null;
+        }
+      }
+    );
+    const indexPatterns = (
+      await Promise.all(promises)
+    ).filter((indexPattern: IndexPattern | null): indexPattern is IndexPattern =>
+      Boolean(indexPattern)
+    );
+    // passing edit url and index patterns to the output of this embeddable for
+    // the container to pick them up and use them to configure filter bar and
+    // config dropdown correctly.
+    const input = this.getInput();
+    const title = input.hidePanelTitles
```

Comment:
could this double-tertiary statement be rewritten to something more readable? 🙏

Reply:
Yeah, that was not super nice looking. Fixed it up a little!